### ISSUE TITLE
Request to be able to use hardware acceleration with a proper docker runtime.

### ISSUE THREAD

**Issue by sudo-kraken** [photoview/photoview#652]
**Is your feature request related to a problem? Please describe.**
I think we would all benefit of being able to offload workloads to the gpu ideally nvidia by adding nvenc.

**Describe the solution you'd like**
Adding the capability to use nvenc_264 and the environment variables as below
      NVIDIA_VISIBLE_DEVICES: all
      NVIDIA_DRIVER_CAPABILITIES: compute,video,utility

Thank you very much.

**Comment by kkovaletp** [comment]
I'm working on it: found the https://hub.docker.com/r/linuxserver/ffmpeg and trying to make it compatible with Photoview and integrate it as an external video converter.
Also, found a list of settings and dependencies to let Darktable to utilize GPU as well.

**Comment by kkovaletp** [comment]
I've finished my PoC for this feature in my forked repo https://github.com/kkovaletp/photoview/pull/26
There might be some more time before it gets ported and merged here, so if you'd like to touch it and play with it - feel free)

BTW, I'll not be able to test the GPU HW acceleration, as I don't have an appropriate environment for that. So, this type of testing should be done by somebody else

**Comment by kkovaletp** [comment]
My implementation was rejected by other revievers, so it won't be ported to the main Photoview repo. Instead, another implementation is going to be developed with no ETA as of now.
Until that new implementation isn't provided, this feature will be available on my forked repo exclusively: https://github.com/kkovaletp/photoview/ - feel free to check the Readme of that repo to get more details and configuration hints

**Comment by googollee** [comment]
Copied from #1001:

> I'm working on [a branch](https://github.com/googollee/photoview/tree/jellyfin-ffmpeg) to enable the hardware acceleration. You can run it with my image `ghcr.io/googollee/photoview:sha-fffd748` (current build) or `ghcr.io/googollee/photoview:jellyfin-ffmpeg` (always the latest build) and set a new environment as `PHOTOVIEW_VIDEO_ENCODER=h264_vaapi` as well as mounting the device `/dev/dri` to see if it works well on your environment.
>
> My branching image is built based on the head of the master repo. I verified it works well with Intel QSV with the environment `PHOTOVIEW_VIDEO_ENCODER=h264_qsv`.

I think this branch should work with AMD. @Uatschitchun, it'd be good if you can report it in your environment.

But Nvidia is more complex. It requires [NVIDIA Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html).

**Comment by Uatschitchun** [comment]
@googollee I'm running on intel iGPU, but gonna try your branch asap.

Maybe a pointer in the docker-compose examples would be nice regarding devices & hw_accel?

**Comment by googollee** [comment]
> Maybe a pointer in the docker-compose examples would be nice regarding devices & hw_accel?

I'm still figuring how to write a proper doc about enabling hardware acceleration. Intel/AMD/Nvidia have different way to set up the docker.

**Comment by DaCHack** [comment]
@kkovaletp  @googollee : Any idea when Intel HW-acceleration will be available in the main repo? I am running release 2.4 and will be happy to test on an Intel iGPU.

**Comment by kkovaletp** [comment]
@DaCHack, I hope that once the #1056 is merged, it should work in the `master` image

**Comment by kkovaletp** [comment]
@googollee, I guess that this issue is also implemented by the recent merges, right? Or do we have still missing parts here?

**Comment by googollee** [comment]
The feature is done. I think we can close it. If we have other problem, it should be a new issue.

**Comment by DaCHack** [comment]
> @DaCHack, I hope that once the #1056 is merged, it should work in the `master` image

Oh great! Any ETA for the next release or is it just waiting for the backlog on the roadmap to be completed?

Just checking whether I move from 2.4 to Master or just wait.

**Comment by kkovaletp** [comment]
> Any ETA for the next release or is it just waiting for the backlog on the roadmap to be completed?

Yes, basically, once all (or most of) the tickets in the "Before next release" column are done, we're going to make a release. There is no ETA as it is difficult to plan.

I'd recommend you try the feature from `master` and let us how it works for you. Especially, if you'd want to configure not QSV protocol, as we don't have an environment to test other than QSV protocols. So, if you find some issues, we might be able to fix them before the release.
It is up to you whether to switch your main instance or set up a new testing one. But the 2nd option looks safer from a general perspective.
